### PR TITLE
add AccountSecurity component

### PR DESCRIPTION
**Pull request by erikphansen**
## Description
This PR fleshes out the Account Security section of Profile 2.0

**TODO:**
- [ ] add tests for the AccountSecurityContent component (these are mostly done locally, but I wanted to get this PR open)

## Testing done
Local + pretty solid unit tests for the new components

## Screenshots
<details>
  <summary>All green</summary>
  
image

</details>

<details>
  <summary>Action needed</summary>
  
image

</details>

<details>
  <summary>Case where MHV section is not shown at all</summary>
  
image

</details>

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/personalization/profile-2/components/MHVTermsAndConditionsStatus.jsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+};
+
+/**
+ * A static helper so that potential parent components will know if this
+ * component will render content at all and potentially adjust their own
+ * rendering logic to account for that. For example, if the parent component
+ * will wrap this component in other markup and it wants to avoid making that
+ * additional wrapper markup if this component will not render anything.
+ */
+MHVTermsAndConditionsStatus.willRenderContent = mhvAccount =>
```
Comment (erikphansen) [inline src/applications/personalization/profile-2/components/MHVTermsAndConditionsStatus.jsx:69]: This is the only "weird" thing about this PR. I wanted a way for the parent `AccountSecurityContent` component will know if it needs to add this component to a table view or not. The parent could do this same check itself, but it felt like the logic to determine if this `MHVTermsAndConditionsStatus` component would render anything should live with the component itself. Is there a better way to do this? Could the parent do a sort of pre-render of this component and if it actually rendered anything then it could be added to the table view?

File: src/applications/personalization/profile-2/components/AccountSecurityContent.jsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      ),
+    },
+  ];
+
+  if (showMHVTermsAndConditions) {
+    data.push({
+      title: 'Terms and conditions',
+      value: <MHVTermsAndConditionsStatus mhvAccount={mhvAccount} />,
+    });
+  }
```
Comment (micahchiang) [inline src/applications/personalization/profile-2/components/AccountSecurityContent.jsx:50]: This is cool. Yeah I see what you were saying about just keeping the static method on `MHVTermsAndConditionsStatus`. Makes sense to me.

**Review by micahchiang — APPROVED** [review]
lgtm!

File: src/applications/personalization/profile-2/components/AccountSecurityContent.jsx (hunk 2 above)
Comment (ncksllvn) [inline src/applications/personalization/profile-2/components/AccountSecurityContent.jsx:50]: This `data` array looked a little funky to me at first. Instead of constructing the data that way, you could have `ProfileInfoTable` also export a `ProfileInfoTableRow` and do something like -

```jsx
<ProfileInfoTable fieldName="accountSecurity">
    <ProfileInfoTableRow 
        title="Identity verification" 
        value={<IdentityVerificationStatus isIdentityVerified={isIdentityVerified} />
    <ProfileInfoTableRow 
        title="Terms and conditions" 
        value={<MHVTermsAndConditionsStatus mhvAccount={mhvAccount} />
</ProfileInfoTable>
```

It might be overkill for this, but I've been wishing our MegaMenu component was designed with components as building blocks instead of processing a complicated data structure.

File: src/applications/personalization/profile-2/components/MHVTermsAndConditionsStatus.jsx (hunk 1 above)
Comment (ncksllvn) [reply src/applications/personalization/profile-2/components/MHVTermsAndConditionsStatus.jsx:69]: Hm, I see what you mean. I suppose it could be exported as a standalone selector? Or if it returns `null` anyway from the `render` if does have the right conditions, maybe the table could just filter it?

File: src/applications/personalization/profile-2/components/TwoFactorAuthorizationStatus.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      <Verified>
+        You’ve added an extra layer of security to your account with 2-factor
+        authentication.
+      </Verified>
+    );
+  }
+
+  const mfaHandler = useSSO => {
+    recordEvent({ event: 'multifactor-link-clicked' });
+    mfa(useSSO ? 'v1' : 'v0');
```
Comment (ncksllvn) [inline src/applications/personalization/profile-2/components/TwoFactorAuthorizationStatus.jsx:20]: Not part of this PR work, but if you find yourself checking `useSSO`, we should ask that team to create a layer of abstraction somehow because these floating strings seem frail.

File: src/applications/personalization/profile-2/components/TwoFactorAuthorizationStatus.jsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  return (
+    <>
+      <p className="vads-u-margin--0">
+        Add an extra layer of security (called 2-factor authentication). This
+        helps to make sure only you can access your account - even if someone
+        gets your password.
+      </p>
+      <p className="vads-u-margin-bottom--0">
+        <button className="va-button-link" onClick={() => mfaHandler(useSSOe)}>
```
Comment (ncksllvn) [inline src/applications/personalization/profile-2/components/TwoFactorAuthorizationStatus.jsx:31]: Since `useSSOe` is in the same scope anyway, you could omit the argument.

File: src/applications/personalization/profile-2/components/Verified.jsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Verified = ({ children }) => (
+  <p className="vads-u-margin--0 vads-u-padding-left--3">
+    <i className="fa fa-check vads-u-color--green vads-u-margin-right--0p5 vads-u-margin-left--neg3" />{' '}
```
Comment (ncksllvn) [inline src/applications/personalization/profile-2/components/Verified.jsx:6]: Interesting - where's the icon patrol bot?

**Review by ncksllvn — APPROVED** [review]
Looks great!

File: src/applications/personalization/profile-2/components/Verified.jsx (hunk 5 above)
Comment (erikphansen) [reply src/applications/personalization/profile-2/components/Verified.jsx:6]: 🤷

File: src/applications/personalization/profile-2/components/TwoFactorAuthorizationStatus.jsx (hunk 3 above)
Comment (erikphansen) [reply src/applications/personalization/profile-2/components/TwoFactorAuthorizationStatus.jsx:20]: @ericbuckley @hinzed1127 might have thoughts about that? I believe one of those guys set up this system.

File: src/applications/personalization/profile-2/components/MHVTermsAndConditionsStatus.jsx (hunk 1 above)
Comment (erikphansen) [reply src/applications/personalization/profile-2/components/MHVTermsAndConditionsStatus.jsx:69]: Hmm, the standalone selector sounds like the alternative I like best. I tried googling to see if my approach was something other people did or if it was some kind of anti-pattern or what. It feels like this is something React could support _somehow_. Simply duplicating logic that already exists in the render function just feels wrong :( I can't imagine I'm the only person who has run into need before.

Comment (ncksllvn) [reply src/applications/personalization/profile-2/components/MHVTermsAndConditionsStatus.jsx:69]: It's definitely a valid separation of concerns. I don't have any issue with it being this way really. I'd maybe bind it to class as a `static` function because the `static` keyword is fun.

File: src/applications/personalization/profile-2/components/AccountSecurityContent.jsx (hunk 2 above)
Comment (erikphansen) [reply src/applications/personalization/profile-2/components/AccountSecurityContent.jsx:50]: I definitely like this idea for a future improvement. This seems more React-y. And this would make it easier to add support for different kinds of table rows.

File: src/applications/personalization/profile-2/components/TwoFactorAuthorizationStatus.jsx (hunk 3 above)
Comment (ericbuckley) [reply src/applications/personalization/profile-2/components/TwoFactorAuthorizationStatus.jsx:20]: @ncksllvn this is something I added a couple of weeks ago and part of what the VA Startup team is working on for the SSOe changes.  What do you mean by "floating strings"?

File: src/applications/personalization/profile-2/components/MHVTermsAndConditionsStatus.jsx (hunk 1 above)
Comment (erikphansen) [reply src/applications/personalization/profile-2/components/MHVTermsAndConditionsStatus.jsx:69]: well this isn't a `class` so this is the best I can do :-D

Comment (erikphansen) [reply src/applications/personalization/profile-2/components/MHVTermsAndConditionsStatus.jsx:69]: Nice that Sophie, who used to be on the core React team I believe, assures me that I'm not crazy in thinking there isn't really a great way to determine if a component will render https://stackoverflow.com/a/39060039/585275

File: src/applications/personalization/profile-2/components/Verified.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (ncksllvn) [reply src/applications/personalization/profile-2/components/Verified.jsx:6]: @erikphansen @kelsonic and I are designing a component that will hide your icons from the bot - https://github.com/department-of-veterans-affairs/vets-website/pull/12212#discussion_r408247824. 

```
function ConcealedIcon({ className }) {
   // The bot looks for an <i> element, so we have to do it this way
    const i = document.createElement('i');
    i.className = className;
   
  // Dynamically construct the prop so we don't have to disable the linter
   const prop = ['dangerously', 'Set', 'Inner', 'Html'];
   const props = {
       [prop.join('')]: { __html: i.outerHtml },
   };
   
   return <span {...props} />
}
```

LMK if this works for you or if you have any ideas on how to improve

File: src/applications/personalization/profile-2/components/TwoFactorAuthorizationStatus.jsx (hunk 3 above)
Comment (ncksllvn) [reply src/applications/personalization/profile-2/components/TwoFactorAuthorizationStatus.jsx:20]: I was referring to the `'v1'` and `'v0'` values. Totally just brainstorming, but it would be neat if these were abstracted somehow so that any logic functions apply it consistently. As in, maybe app state stores the login method so that `mfa()` applies it automatically, or just as constants somehow to add some meaning to `v0` vs. `v1`.

File: src/applications/personalization/profile-2/components/Verified.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (erikphansen) [reply src/applications/personalization/profile-2/components/Verified.jsx:6]: I do wonder why the bot didn't find the icon when I opened the PR

[the remark this answers is not in the document]
Comment (rianfowler) [reply src/applications/personalization/profile-2/components/Verified.jsx:6]: cc @bkjohnson

[the remark this answers is not in the document]
Comment (ncksllvn) [reply src/applications/personalization/profile-2/components/Verified.jsx:6]: I'm just being obnoxious @rianfowler lol

[the remark this answers is not in the document]
Comment (bkjohnson) [reply src/applications/personalization/profile-2/components/Verified.jsx:6]: The bot only runs on PRs, so in this case the commit that added the icon was already pushed to the branch but it didn't have a PR associated when the bot began checking.  The follow up commits to the PR are what triggered the comment.

Any feedback on how to make the bot less annoying is welcome.
